assets/src/edit-story/components/canvas/gridview/useGridViewKeys.js

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useMemo, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useKeyDownEffect } from '../../keyboard';
+import { useStory } from '../../../app';
+
+function useGridViewKeys(ref, gridRef, pageRefs, isRTL) {
+  const { arrangePage, currentPageId, pages } = useStory(
+    ({ state: { pages, currentPageId }, actions: { arrangePage } }) => ({
+      arrangePage,
+      currentPageId,
+      pages,
+    })
+  );
+
+  const [focusedPageId, setFocusedPageId] = useState(currentPageId);
+
+  const pageIds = useMemo(() => pages.map(({ id }) => id), [pages]);
+
+  // Navigate focus left, right, up, down
+  useKeyDownEffect(
+    ref,
+    { key: ['up', 'down', 'left', 'right'] },
+    ({ key }) => {
+      switch (key) {
+        case 'ArrowLeft':
+        case 'ArrowRight': {
+          const dir = getArrowDir(key, 'ArrowRight', 'ArrowLeft', isRTL);
+
+          if (dir === 0) {
+            return;
+          }
+
+          const index = pageIds.indexOf(focusedPageId);
+          const nextIndex = index + dir;
+
+          if (nextIndex < 0 || nextIndex === pageIds.length) {
+            return;
+          }
+
+          const pageId = pageIds[nextIndex];
+
+          setFocusedPageId(pageId);
+
+          const page = pageRefs.current && pageRefs.current[pageId];
+
+          focusOnPage(page);
+
+          break;
+        }
+        case 'ArrowUp':
+        case 'ArrowDown': {
```

Comment:
nit: constants for key names to reference easier in logic with you're matching keys like line 79